src/main/java/com/github/davidmoten/rtree/internal/Functions.java

```diff
@@ -9,30 +9,15 @@ private Functions() {
     }
     
     public static <T> Func1<T, T> identity() {
```

Comment:
Now that you've drawn my attention to them, these methods should return singletons. I prefer holder pattern if you can be bothered to change them.

Reply:
I agree, but for generic I cannot create static generic field like ```private static final Func1<T, T> IDENTITY = t -> t```
Did you think something like this?

Please check solution of jdk: ```java.util.function.Function.identity()```

Reply:
Would be interesting to benchmark. I would expect that this would involve less allocation cost:

```java
    // actually use a holder pattern here
    private static final Func1<Object, Object> IDENTITY = t -> t;
    
    @SuppressWarnings("unchecked")
    public static final <T> Func1<T,T> identity() {
        return (Func1<T, T>) IDENTITY;
    }
```
At runtime no mucking around with generic types occurs so the cast would be to its already declared type which should be a no-op. Of course benchmarks are the real decider here.

Reply:
BTW we don't need a benchmark of this fact added to this project. You can add it temporarily for your own tests but don't commit. (if you are going to benchmark of course).